File: android/app/src/main/kotlin/com/xpx/vault/ui/setup/FirstLaunchRouter.kt
package com.xpx.vault.ui.setup

import android.content.Context
import com.xpx.vault.data.db.PhotoVaultDatabase
import com.xpx.vault.ui.backup.ExternalBackupLocation

/**
 * 首启分叉路由：根据本机 SecuritySetting 与外部 backup.dat 判断应走哪条路径。
 *
 * - [Branch.Fresh]：从未设置 PIN 且外部无备份 → 走原有 SETUP_ENTER 新建流程。
 * - [Branch.RestoreLogin]：从未设置 PIN 但外部有 backup.dat → 走「输入密码 = 恢复凭证 + 本机新密码」分支。
 * - [Branch.Unlock]：已有 SecuritySetting → 走常规 UNLOCK 流程。
 */
object FirstLaunchRouter {
    sealed class Branch {
        object Fresh : Branch()
        object RestoreLogin : Branch()
        object Unlock : Branch()
    }

    suspend fun detect(context: Context, db: PhotoVaultDatabase): Branch {
        val setting = db.securitySettingDao().getById()
        if (setting != null) return Branch.Unlock
        val hasAutoPackage = ExternalBackupLocation.findAuto(context) != null
        return if (hasAutoPackage) Branch.RestoreLogin else Branch.Fresh
    }
}

File: ios/LumaNox/Core/Setup/FirstLaunchRouter.swift
import Foundation

/// 首启分叉 — 对齐 Android [FirstLaunchRouter]。
enum FirstLaunchBranch: Equatable {
    /// 从未设置 PIN，外部无 `backup.dat`
    case fresh
    /// 从未设置 PIN，但已授权目录中存在 `backup.dat`
    case restoreLogin
    /// 已有 PIN
    case unlock
}

enum FirstLaunchRouter {
    static func detect(securityStore: SecuritySettingsStore = .shared) -> FirstLaunchBranch {
        securityStore.reload()
        if securityStore.hasPinConfigured {
            return .unlock
        }
        if ExternalBackupLocation.findAutoBackup() {
            return .restoreLogin
        }
        return .fresh
    }
}
